src/utils/presence.js

```diff
@@ -113,8 +126,38 @@ export const statusFromPresence = (presence: UserPresence | void): PresenceStatu
   return presence.aggregated.status;
 };
 
+// TODO(server-6.0): Remove; UserStatus['away'] was deprecated at FL 148.
 export const statusFromPresenceAndUserStatus = (
   presence: UserPresence | void,
   userStatus: UserStatus,
 ): PresenceStatus | 'unavailable' =>
   userStatus.away ? 'unavailable' : statusFromPresence(presence);
+
+/**
+ * Get a user's overall presence status, aggregated from all their devices.
+ *
+ * Gives null when we're missing data to determine this, such as when the
+ * user doesn't exist.
+ */
+export const getPresenceStatusForUserId = (
+  state: PerAccountState,
+  userId: UserId,
+): PresenceStatus | 'unavailable' | null => {
+  const presence = getPresence(state);
+  const user = tryGetUserForId(state, userId);
+  if (!user) {
+    return null;
+  }
+  const userPresence = (presence[user.email]: UserPresence | void);
+  const userStatus = getUserStatus(state, userId);
+  if (!userPresence || !userPresence.aggregated) {
+    return null;
+  }
```

Comment:
nit: swap order so the `userPresence` checks and early return are immediately after defining `userPresence`

(The old code didn't do this because it was a series of `useSelector` calls, and so the control flow had to be a bit contorted to put the conditional after all of those. But here, we can.)